pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -134,6 +146,183 @@ func (m *KubeIPResolverInstance) EnrichEvent(ev any) error {
+
+func (m *KubeIPResolverInstance) PreStart(gadgetCtx operators.GadgetContext) error {
+	m.k8sInventory.Start()
+	return nil
+}
+
+func (m *KubeIPResolverInstance) Start(gadgetCtx operators.GadgetContext) error {
+	for ds, acc := range m.endpointsAccessors {
+		for _, a := range acc {
+			ds.Subscribe(func(source datasource.DataSource, data datasource.Data) error {
+				ips := a.root.GetSubFieldsWithTag("type:" + ipAddrType)
+				if len(ips) != 1 {
+					return fmt.Errorf("expected %d %q field, got %d", 1, ipAddrType, len(ips))
+				}
+				if ips[0].Size() != 16 {
+					return fmt.Errorf("expected %q to have 16 bytes, got %d", ipAddrType, ips[0].Size())
+				}
+
+				version := a.root.GetSubFieldsWithTag("name:version")
+				if len(version) != 1 {
+					return fmt.Errorf("expected %d %q field, got %d", 1, "version", len(version))
+				}
+
+				ip := ips[0].Get(data)
+				if ip == nil {
+					return fmt.Errorf("expected %q to have a value", ipAddrType)
+				}
+				v, err := version[0].Uint8(data)
+				if err != nil {
+					return fmt.Errorf("expected %q to have a value", "version")
+				}
+				var addrStr string
+				switch v {
+				case 4:
+					addrStr = net.IP(ips[0].Get(data)[:4]).String()
+				case 6:
+					addrStr = net.IP(ips[0].Get(data)).String()
+				default:
+					return fmt.Errorf("unsupported IP version %d", v)
+				}
+				pod := m.k8sInventory.GetPodByIp(addrStr)
+				if pod != nil {
+					if pod.Spec.HostNetwork {
+						return nil
+					}
+					a.subK8sName.Set(data, []byte(pod.Name))
+					a.subK8sKind.Set(data, []byte("pod"))
+					a.subK8sNamespace.Set(data, []byte(pod.Namespace))
+					// TODO: labels should be a map/slice
+					var labels []string
+					for k, v := range pod.Labels {
+						labels = append(labels, fmt.Sprintf("%s=%s", k, v))
+					}
+					a.subK8sLabels.Set(data, []byte(strings.Join(labels, ",")))
+					return nil
+				}
+
+				svc := m.k8sInventory.GetSvcByIp(addrStr)
+				if svc != nil {
+					a.subK8sKind.Set(data, []byte("svc"))
+					a.subK8sName.Set(data, []byte(svc.Name))
+					a.subK8sNamespace.Set(data, []byte(svc.Namespace))
+					// TODO: labels should be a map/slice
+					var labels []string
+					for k, v := range svc.Labels {
+						labels = append(labels, fmt.Sprintf("%s=%s", k, v))
+					}
+					a.subK8sLabels.Set(data, []byte(strings.Join(labels, ",")))
+					return nil
+				}
+
+				return nil
+			}, 0)
+		}
+	}
+	return nil
+}
+
+func (m *KubeIPResolverInstance) PostStop(gadgetCtx operators.GadgetContext) error {
+	m.k8sInventory.Stop()
+	return nil
+}
+
+func (m *KubeIPResolverInstance) Stop(gadgetCtx operators.GadgetContext) error {
+	return nil
+}
+
 func init() {
 	operators.Register(&KubeIPResolver{})
+	// TODO: use localmanager.EnrichWithK8sApiserver instead to control this for ig,
+	// once we support sharing params. For now, we only support this operator in Kubernetes.
+	if environment.Environment == environment.Kubernetes {
+		operators.RegisterDataOperator(&KubeIPResolver{})
+	}
```

Comment:
If you don't want to support it on `ig` for now, simply don't import it in `cmd/ig/main.go`.

Reply:
I opened https://github.com/inspektor-gadget/inspektor-gadget/issues/3603 to track this!